### PR TITLE
Adjust hydra syncer on evaluation profile

### PR DESCRIPTION
**Pull request by cnvergence**
See also https://github.com/kyma-incubator/reconciler/issues/961 https://github.com/kyma-project/kyma/issues/13498

### CODE REVIEW

**Comment by cnvergence** [comment]
- deploy Kyma on 2.0.4
- deploy oauth2client
- upgrade to Kyma main 
- check /clients on Ory Hydra, if they are empty after reconciliation

File: pkg/reconciler/instances/ory/hydra/syncer_test.go [hunk 1]
```diff
@@ -62,51 +63,48 @@ func Test_TriggerSynchronization(t *testing.T) {
...
-		hydraStartTimePod3 := time.Date(2500, 10, 10, 10, 10, 6, 10, time.UTC)
+		hydraStartTimePod := time.Date(2022, 10, 10, 10, 10, 7, 10, time.UTC)
+		hydraStartTimePod2 := time.Date(2022, 10, 10, 10, 10, 6, 10, time.UTC)
 		hydraMasesterPodStartTime := time.Date(2022, 10, 10, 10, 10, 6, 10, time.UTC)
 		kubeclient := fakeClient()
-		addPod(kubeclient, "hydra1", "hydra", hydraStartTimePod1, t, v1.PodRunning)
+		addPod(kubeclient, "hydra1", "hydra", hydraStartTimePod, t, v1.PodRunning)
 		addPod(kubeclient, "hydra2", "hydra", hydraStartTimePod2, t, v1.PodRunning)
-		addPod(kubeclient, "hydra3", "hydra", hydraStartTimePod3, t, v1.PodPending)
 		createDeployment(kubeclient, hydraMasesterPodStartTime, t)
```
Comment (piotrkpc) [inline pkg/reconciler/instances/ory/hydra/syncer_test.go:77]: maybe we can do some renaming to improve readability?
```suggestion
		hydraStartTimeEarlier := time.Date(2022, 10, 10, 10, 10, 7, 10, time.UTC)
		hydraStartTimeLater := hydraStartTimeEarlier.Add(time.Second)
		hydraMasesterPodStartTime := hydraStartTimeEarlier
		kubeclient := fakeClient()
		addPod(kubeclient, "hydra1", "hydra", hydraStartTimeEarlier, t, v1.PodRunning)
		addPod(kubeclient, "hydra2", "hydra", hydraStartTimeLater, t, v1.PodRunning)
```

**Comment by piotrkpc** [comment]
test log:
```
pkopec@piotr-nb kyma-incubator/reconciler (ory-syncer %) » kyma provision k3d --port 8080:80@loadbalancer --ci                                                         
  Checking if port flags are valid
  Checking if k3d registry of previous kyma installation exists
  Checking if k3d cluster of previous kyma installation exists
  Deleted k3d registry of previous kyma installation
- k3d status verified
- Created k3d registry 'kyma-registry:5001'
- Created k3d cluster 'kyma'
pkopec@piotr-nb kyma-incubator/reconciler (ory-syncer %) » make build-linux                                    
CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o ./bin/reconciler-linux -ldflags '-s -w' ./cmd/reconciler
CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o ./bin/mothership-linux -ldflags '-s -w' ./cmd/mothership
pkopec@piotr-nb kyma-incubator/reconciler (ory-syncer %) » ./bin/mothership-linux local  --version 2.0.4 --components cluster-essentials,istio-configuration,ory --profile evaluation 
[...]
pkopec@piotr-nb kyma-incubator/reconciler (ory-syncer %) » k exec -it -n kyma-system ory-hydra-765b558996-n8fq5  -- hydra clients list --endpoint http://127.0.0.1:4445/ 
| CLIENT ID | NAME | RESPONSE TYPES | SCOPE | REDIRECT URIS | GRANT TYPES | TOKEN ENDPOINT AUTH METHOD |
|-----------|------|----------------|-------|---------------|-------------|----------------------------|
pkopec@piotr-nb kyma-incubator/reconciler (ory-syncer %) » export NAMESPACE=default                                                                                     
pkopec@piotr-nb kyma-incubator/reconciler (ory-syncer %) » export CLIENT_NAME=test                                        
pkopec@piotr-nb kyma-incubator/reconciler (ory-syncer %) » cat <<EOF | kubectl apply -f -                                                                                                                          
apiVersion: hydra.ory.sh/v1alpha1
kind: OAuth2Client
metadata:
  name: $CLIENT_NAME
  namespace: $NAMESPACE
spec:
  grantTypes:
    - "client_credentials"
  scope: "read write"
  secretName: $CLIENT_NAME
EOF
oauth2client.hydra.ory.sh/test created
pkopec@piotr-nb kyma-incubator/reconciler (ory-syncer %) » k exec -it -n kyma-system ory-hydra-765b558996-n8fq5  -- hydra clients list --endpoint http://127.0.0.1:4445/
|              CLIENT ID               | NAME | RESPONSE TYPES |   SCOPE    | REDIRECT URIS |    GRANT TYPES     | TOKEN ENDPOINT AUTH METHOD |
|--------------------------------------|------|----------------|------------|---------------|--------------------|----------------------------|
| d7b4adfd-9724-4830-8850-6ac484a25005 |      |                | read write |               | client_credentials | client_secret_basic        |
pkopec@piotr-nb kyma-incubator/reconciler (ory-syncer %) » ./bin/mothership-linux local  --version main --components cluster-essentials,istio,ory --profile evaluation
[...]
pkopec@piotr-nb kyma-incubator/reconciler (ory-syncer %) » k exec -it -n kyma-system ory-hydra-765b558996-lkglv -- hydra clients list --endpoint http://127.0.0.1:4445/                                                             1 ↵
|              CLIENT ID               | NAME | RESPONSE TYPES |   SCOPE    | REDIRECT URIS |    GRANT TYPES     | TOKEN ENDPOINT AUTH METHOD |
|--------------------------------------|------|----------------|------------|---------------|--------------------|----------------------------|
| d7b4adfd-9724-4830-8850-6ac484a25005 |      |                | read write |               | client_credentials | client_secret_basic        |
```

**Review by piotrkpc — APPROVED** [review]
/meow

**Pull request merged**